burn/Cargo.toml

```diff
@@ -17,10 +17,14 @@ std = [
   "burn-core/std",
 ]
 train = ["std", "burn-train"] # Training requires std
+dataset = ["burn-dataset"]
+dataset-sqlite = ["burn-dataset/sqlite"]
+dataset-sqlite-bundled = ["burn-dataset/sqlite-bundled"]
```

Comment:
Sorry, I'm a bit confused here. Isn't the goal for users to be able to import the 'burn' crate and not need to import the subcrates? Wouldn't moving these flags into burn-core require the user to also import burn-core so they can specify features on it - and wouldn't it be easier for them to just import dataset directly in that case? Or are you suggesting that I put these flags on both burn and burn-core, and have burn feed them to burn-core which in turn feeds them to burn-dataset? Or did you have something else in mind that I've misunderstood?

Reply:
`burn-core` already have the `burn-dataset` dependency, it's already being re-exported there, so we need to connect the dots an bring those features flags to `burn-core` as well. Since there are two levels of re-export, `burn` should just pass those feature flags to `burn-core`.

Reply:
Ok, thanks. Is the latest commit what you had in mind?